app/lib/actions/katello/content_view/publish.rb

```diff
@@ -6,10 +6,10 @@ class Publish < Actions::EntryAction
         include ::Katello::ContentViewHelper
         attr_accessor :version
         # rubocop:disable Metrics/MethodLength,Metrics/AbcSize,Metrics/CyclomaticComplexity
-        def plan(content_view, description = "", options = {})
+        def plan(content_view, description = "", options = {importing: false})
```

Comment:
This importing option is unnecessary imo. why not just do content_view.import_only?

Reply:
We need a way to identify if the publish action was called directly (via API for example) vs the CV import action in order to know whether we should allow the publish of an import_only CV. Lemme know if I'm missing the point though

Reply:
I might be missing something here.  In both cases the thing will fail if you are trying publish if your cv is import_only (I see that you already do this in the model?)
I dont know why it should matter how the import call was made..

Reply:
If we don't have a flag like this, how do we allow publishing an import_only CV through CV import, but deny the same publish when performed through the ordinary publish API? Seems like we need to know when we are performing an import

Reply:
Ah good point. That makes sense.Lets keep it your way.